### PR TITLE
Adapt uvisor-lib to uVisor refactoring

### PR DESCRIPTION
**Pull request by AlessandroA**
The updated `uvisor-lib` now features:

* Platform-agnostic `CMakeLists.txt`
    * The device-family-specific configurations are picked from a `$family.cmake` file
* Updated binaries with the new binary naming schema: `$family_$configuration.o`

@Patater 
@meriac

### CODE REVIEW

File: source/CMakeLists.txt [hunk 1]
```diff
@@ -17,27 +17,35 @@
...
-        set(UVISOR_DIR "MK64FN1M0XXX12")        # FRDM-K64F
+# Determine the uVisor binary based on the target.
+# Note: We currently only support the GCC toolchain.
+set(UVISOR_SUPPORTED false)
+if(TARGET_LIKE_GCC)
+    # Pick the correct folder based on the target family.
+    if(TARGET_LIKE_KINETIS_K64_GCC)
+        set(UVISOR_FAMILY "kinetis")
+    elseif(TARGET_LIKE_STM32)
+        set(UVISOR_FAMILY "stm32")
```
Comment (Patater) [inline source/CMakeLists.txt]: Is there a way to error from CMake if the target is not like kinetis or stm32?

File: source/CMakeLists.txt [hunk 2]
```diff
@@ -17,27 +17,35 @@
...
     endif()
-elseif(TARGET_LIKE_STM32F429I_DISCO)
-    if(TARGET_LIKE_GCC)
-        set(UVISOR_PRESENT true)
-        set(UVISOR_DIR "STM32F429xx")           # STM32F429I
+
+    # Infer path to binary.
+    # The included file defines UVISOR_CONFIGURATION.
+    include(${UVISOR_FAMILY}/get_configuration.cmake)
```
Comment (Patater) [inline source/CMakeLists.txt]: Would be nice to reach into the uvisor submodule here instead (when we get there, I guess).

**Comment by Patater** [comment]
Does this mean that UVISOR_PRESENT has gone away? Do we need to change anything in yotta (perhaps YOTTA_CFG_UVISOR_PRESENT)?

File: source/CMakeLists.txt (hunk 1 above)
Comment (AlessandroA) [inline source/CMakeLists.txt]: We should not throw an error. `UVISOR_SUPPORTED` is pessimistically set to `false` and updated to `true` if the target platform is either `kinetis` or `stm32` AND there exists a uVisor configuration for the specific yotta target.

So if `UVISOR_SUPPORTED` remain `false` uVisor is left out and a warning is thrown. That warning actually becomes an error if the target mistakenly stated that uVisor is present (`YOTTA_CFG_UVISOR_PRESENT`).

File: source/CMakeLists.txt (hunk 2 above)
Comment (AlessandroA) [inline source/CMakeLists.txt]: Yup! :)

File: source/CMakeLists.txt (hunk 1 above)
Comment (Patater) [inline source/CMakeLists.txt]: Good good. I'm glad that the warning and error stuff still works as intended.

**Comment by AlessandroA** [comment]
@Patater Nope, that's part of the reason for the change: `YOTTA_CFG_UVISOR_PRESENT` and `UVISOR_PRESENT` in this file had nothing to do with each other (i.e. they were set independently). The variable is only used internally to choose which files to include in the build process. The name plays well with the logic:

```
if UVISOR_SUPPORTED
    build all
else
    build unsupported.cpp
end
```

**Comment by Patater** [comment]
Yes. I agree. Much clearer.

Since all the warning and error things related to UVISOR_SUPPORTED and UVISOR_PRESENT and YOTTA_CFG_UVISOR_PRESENT are still working, I'm good with these changes.

LGTM

**Comment by meriac** [comment]
+1

**Pull request merged**